lms/djangoapps/instructor/views/api.py

```diff
@@ -2625,24 +2628,64 @@ def enable_certificate_generation(request, course_id=None):
     return redirect(_instructor_dash_url(course_key, section='certificates'))
 
 
-#---- Gradebook (shown to small courses only) ----
-@cache_control(no_cache=True, no_store=True, must_revalidate=True)
-@require_level('staff')
-def spoc_gradebook(request, course_id):
+def calculate_page_info(offset, total_students):
     """
-    Show the gradebook for this course:
-    - Only shown for courses with enrollment < settings.FEATURES.get("MAX_ENROLLMENT_INSTR_BUTTONS")
-    - Only displayed to course staff
+    Takes care of sanitizing the offset of current page also calculates offsets for next and previous page
+    and information like total number of pages and current page number.
+
+    :param offset: offset for database query
+    :return: tuple consist of page number, query offset for next and previous pages and valid offset
     """
-    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
-    course = get_course_with_access(request.user, 'staff', course_key, depth=None)
 
+    # validate offset
+    if not (isinstance(offset, int) or offset.isdigit()) or int(offset) < 0 or int(offset) >= total_students:
+        offset = 0
+    else:
+        offset = int(offset)
+
+    next_offset = offset + MAX_STUDENTS_PER_PAGE_GRADE_BOOK
+    previous_offset = offset - MAX_STUDENTS_PER_PAGE_GRADE_BOOK
+    page_num = ((offset / MAX_STUDENTS_PER_PAGE_GRADE_BOOK) + 1)
+
+    if previous_offset < 0 or offset == 0:
+        previous_offset = None  # No previous page
+
+    if next_offset >= total_students:  # No next page
+        next_offset = None
```

Comment:
Our convention in platform is to not put comments on the same line as code. They should appear on their own line, and enhance readability of the code (so more complete sentences, not inscrutable dev notes).